cypress/dockerNode.ts

```diff
@@ -0,0 +1,242 @@
+
+		// Getting latest image
+		console.log('\nChecking running containers... 🔍')
+		const localImage = await docker.listImages({ filters: `{"reference": ["${SERVER_IMAGE}"]}` })
+
+		// Remove old container if exists and not initialized by us
+		try {
+			const oldContainer = docker.getContainer(CONTAINER_NAME)
+			const oldContainerData = await oldContainer.inspect()
+			if (oldContainerData.State.Running) {
+				console.log('├─ Existing running container found')
+				if (localImage[0].Id !== oldContainerData.Image) {
+					console.log('└─ But running container is outdated, replacing...')
+				} else {
+					// Get container's IP
+					console.log('├─ Reusing that container')
+					const ip = await getContainerIP(oldContainer)
+					return ip
+				}
+			} else {
+				console.log('└─ None found!')
+			}
+			// Forcing any remnants to be removed just in case
+			await oldContainer.remove({ force: true })
+		} catch (error) {
+			console.log('└─ None found!')
+		}
+
+		// Starting container
+		console.log('\nStarting Nextcloud container... 🚀')
+		console.log(`├─ Using branch '${branch}'`)
+		console.log(`├─ And binding app '${APP_NAME}' from '${APP_PATH}'`)
+		const container = await docker.createContainer({
+			Image: SERVER_IMAGE,
+			name: CONTAINER_NAME,
+			HostConfig: {
+				Binds: [`${APP_PATH}:/var/www/html/apps/${APP_NAME}`],
+			},
+			Env: [
+				`BRANCH=${branch}`,
+			],
+		})
+		await container.start()
+
+		// Get container's IP
+		const ip = await getContainerIP(container)
+
+		console.log(`├─ Nextcloud container's IP is ${ip} 🌏`)
+		return ip
+	} catch (err) {
+		console.log('└─ Unable to start the container 🛑')
+		console.log(err)
+		stopNextcloud()
+		throw new Error('Unable to start the container')
+	}
+}
+
+/**
+ * Clone viewer
+ */
+export const cloneViewer = async function(branch = process.env.BRANCH || 'master') {
+	console.log('\nCloning Viewer...')
+	const container = docker.getContainer(CONTAINER_NAME)
+	await runExec(container, ['git', 'clone', '--depth', '1', '--branch', branch, 'https://github.com/nextcloud/viewer', '/var/www/html/apps/viewer'], true)
+
+	console.log('└─ Nextcloud is now ready to use 🎉')
+}
+
+/**
+ * Configure Nextcloud
+ */
+export const configureNextcloud = async function() {
+	console.log('\nConfiguring nextcloud...')
+	const container = docker.getContainer(CONTAINER_NAME)
+	await runExec(container, ['php', 'occ', '--version'], true)
+
+	// Be consistent for screenshots
+	await runExec(container, ['php', 'occ', 'config:system:set', 'default_language', '--value', 'en'], true)
+	await runExec(container, ['php', 'occ', 'config:system:set', 'force_language', '--value', 'en'], true)
+	await runExec(container, ['php', 'occ', 'config:system:set', 'default_locale', '--value', 'en_US'], true)
+	await runExec(container, ['php', 'occ', 'config:system:set', 'force_locale', '--value', 'en_US'], true)
+	await runExec(container, ['php', 'occ', 'config:system:set', 'enforce_theme', '--value', 'light'], true)
```

Comment:
Ideally also have `"memcache.local"=>"\OC\Memcache\APCu","hashing_default_password"=>true` for faster runs